project/CelebornBuild.scala

```diff
@@ -0,0 +1,604 @@
+  // val jacksonDatabindVersion = "2.6.7.3"
+  val lz4JavaVersion = "1.4.0"
+  val sparkProjectScalaVersion = "2.11.12"
+  // scalaBinaryVersion
+  // val scalaBinaryVersion = "2.11"
+  val sparkVersion = "2.4.8"
+  val zstdJniVersion = "1.4.4-3"
+}
+
+object Spark30 extends SparkClientProjects {
+
+  val sparkClientProjectPath = "client-spark/spark-3"
+  val sparkClientProjectName = "celeborn-client-spark-3"
+  val sparkClientShadeProjectPath = "client-spark/spark-3-shade"
+  val sparkClientShadeProjectName = "celeborn-client-spark-3-shaded"
+
+  val lz4JavaVersion = "1.7.1"
+  val sparkProjectScalaVersion = "2.12.10"
+
+  val sparkVersion = "3.0.3"
+  val zstdJniVersion = "1.4.4-3"
+}
+
+object Spark31 extends SparkClientProjects {
+
+  val sparkClientProjectPath = "client-spark/spark-3"
+  val sparkClientProjectName = "celeborn-client-spark-3"
+  val sparkClientShadeProjectPath = "client-spark/spark-3-shade"
+  val sparkClientShadeProjectName = "celeborn-client-spark-3-shaded"
+
+  val lz4JavaVersion = "1.7.1"
+  val sparkProjectScalaVersion = "2.12.10"
+
+  val sparkVersion = "3.1.3"
+  val zstdJniVersion = "1.4.8-1"
+}
+
+object Spark32 extends SparkClientProjects {
+
+  val sparkClientProjectPath = "client-spark/spark-3"
+  val sparkClientProjectName = "celeborn-client-spark-3"
+  val sparkClientShadeProjectPath = "client-spark/spark-3-shade"
+  val sparkClientShadeProjectName = "celeborn-client-spark-3-shaded"
+
+  val lz4JavaVersion = "1.7.1"
+  val sparkProjectScalaVersion = "2.12.15"
+
+  val sparkVersion = "3.2.4"
+  val zstdJniVersion = "1.5.0-4"
+}
+
+object Spark33 extends SparkClientProjects {
+
+  val sparkClientProjectPath = "client-spark/spark-3"
+  val sparkClientProjectName = "celeborn-client-spark-3"
+  val sparkClientShadeProjectPath = "client-spark/spark-3-shade"
+  val sparkClientShadeProjectName = "celeborn-client-spark-3-shaded"
+
+  // val jacksonVersion = "2.13.4"
+  // val jacksonDatabindVersion = "2.13.4.2"
+  val lz4JavaVersion = "1.8.0"
+  val sparkProjectScalaVersion = "2.12.15"
+  // scalaBinaryVersion
+  // val scalaBinaryVersion = "2.12"
+  val sparkVersion = "3.3.2"
+  val zstdJniVersion = "1.5.2-1"
+}
+
+object Spark34 extends SparkClientProjects {
+
+  val sparkClientProjectPath = "client-spark/spark-3"
+  val sparkClientProjectName = "celeborn-client-spark-3"
+  val sparkClientShadeProjectPath = "client-spark/spark-3-shade"
+  val sparkClientShadeProjectName = "celeborn-client-spark-3-shaded"
+
+  val lz4JavaVersion = "1.8.0"
+  val sparkProjectScalaVersion = "2.12.17"
+
+  val sparkVersion = "3.4.1"
+  val zstdJniVersion = "1.5.2-5"
+
+  lazy val deps = Seq(
+    // Spark Use `log4j-slf4j2-impl` instead of `log4j-slf4j-impl` in SPARK-40511
+    // to fix the error:
+    // ```
+    //   java.lang.NoSuchMethodError: org.apache.logging.slf4j.Log4jLoggerFactory.<init>(Lorg/apache/logging/slf4j/Log4jMarkerFactory;)V
+    // ```
+    "org.apache.logging.log4j" % "log4j-slf4j2-impl" % "2.19.0" % "test"
```

Comment:
does exclude `log4j-slf4j2-impl` and use `log4j-slf4j-impl` work?

Reply:
it doesn't work

Reply:
if the error message is same, I think it may caused by

1. in Maven, slf4j-api 1.x takes effect
2. in SBT, slf4j-api 2.x takes effect

it may be caused by difference dependencies resolution strategy or configuration